File: University/AdadeKhobS1.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
       Scanner scanner = new Scanner(System.in);

       long number = scanner.nextLong();

       long sum = 0, n = 1, m;
       boolean flag = true;

       while (flag) {
           m = (n*(n+1))/2;
           for (int i = 1; i <= m; i++) {
               if (m % i == 0) {
                   sum++;
               }
           }
           if (number <= sum) {
               System.out.print(m);
               flag = false;
           }
           sum = 0;
           n++;
       }
    }
}

File: University/AdadeKhobS1.rb
class Main
    def self.main()
        scanner =  java.util.Scanner
        number = gets.chomp
        sum = 0
        n = 1
        flag = true
        while (flag)
            m = (n * (n + 1)) / 2
            i = 1
            while (i <= m)
                if (m % i == 0)
                    sum += 1
                end
                i += 1
            end
            if (number <= sum)
                print(m)
                flag = false
            end
            sum = 0
            n += 1
        end
    end
end
Main.main()
